packages/fast-components-react-base/src/tree-view-item/tree-view-item.tsx

```diff
@@ -0,0 +1,491 @@
+    }
+
+    private adjustNestedTreeItemCount(count: 1 | -1): void {
+        if (typeof this.context.adjustNestedTreeItemCount === "function") {
+            this.context.adjustNestedTreeItemCount(count);
+        }
+    }
+
+    /**
+     * Handles the focus event of the root element
+     */
+    private handleFocus = (e: React.FocusEvent<HTMLDivElement>): void => {
+        // Only change focusability if the event is on the node itself
+        // and not a child node
+        if (e.target === e.currentTarget) {
+            this.setState({ focusable: true });
+
+            // Notify the parent TreeView component that a new item has been focused,
+            // and should tree this item as the last-focused item
+            if (typeof this.context.setLastFocused === "function") {
+                this.context.setLastFocused(this.rootElement);
+            }
+        }
+    };
+
+    /**
+     * Handles the blur event on the root element
+     */
+    private handleBlur = (e: React.FocusEvent<HTMLDivElement>): void => {
+        // Ignore the event if the event happened on a child node
+        if (!canUseDOM() || e.target !== e.currentTarget) {
+            return;
+        }
+
+        this.setState({ focusable: false });
+    };
+
+    /**
+     * handles the keydown event of the tree view item
+     */
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        if (e.target !== e.currentTarget) {
+            return;
+        }
+
+        switch (e.keyCode) {
+            case keyCodeArrowLeft:
+                this.handleArrowLeft();
+                break;
+            case keyCodeArrowRight:
+                this.handleArrowRight();
+                break;
+            case keyCodeArrowDown:
+                // Prevent scrolling
+                e.preventDefault();
+                this.focusNextNode(1);
+                break;
+            case keyCodeArrowUp:
+                // Prevent scrolling
+                e.preventDefault();
+                this.focusNextNode(-1);
+                break;
+            case keyCodeEnter:
+                this.handleSelected(e);
+                break;
+            case keyCodeSpace:
+                this.handleSpaceBar();
+                break;
+        }
+    };
+
+    /**
+     * Determines if the item has sub-items and should be collapse/expandable
+     */
+    private hasChildNodes(props: TreeViewItemProps = this.props): boolean {
+        return !!props.children;
+    }
+
+    /**
+     * Determines if the item has sub-items which are selected
+     */
+    private hasSelectedChild(): boolean | void {
+        const currentNode: HTMLElement = this.rootElement.current;
+
+        if (!isHTMLElement(currentNode) || !canUseDOM()) {
+            return null;
+        }
+
+        return !!currentNode.querySelector("[aria-selected='true'");
+    }
+
+    /**
+     * Handles when the left arrow is pressed
+     */
+    private handleArrowLeft(): void {
```

Comment:
Good question - we don't have a loc bug on it currently but I would expect it would

Reply:
I'm not sure - The spec details the following:

> Right arrow:
- When focus is on a closed node, opens the node; focus does not move.
- When focus is on a open node, moves focus to the first child node.
- When focus is on an end node, does nothing.

> Left arrow:
- When focus is on an open node, closes the node.
- When focus is on a child node that is also either an end node or a closed node, moves focus to its parent node.
- When focus is on a root node that is also either an end node or a closed node, does nothing.

I'd prefer to ask this question of the standards body and follow-up rather than assume here.

Reply:
Yea, I think we can take it as a follow-up